### ISSUE TITLE
Lack of Color in git-diff view

### ISSUE THREAD

**Issue by jeff-hykin** [atom/github#2486]
## TLDR:
There's almost certainly an issue about this already, but I was unable to find one.
If you could redirect me to the existing issue and close this I'd appreciate it.

## Summary

Current diff display:
![Screen Shot 2020-07-12 at 4 29 22 PM](https://user-images.githubusercontent.com/17692058/87257870-5d575680-c464-11ea-8858-0e034db84f5c.png)

What was expected:
![Screen Shot 2020-07-12 at 5 25 58 PM](https://user-images.githubusercontent.com/17692058/87257947-f7b79a00-c464-11ea-93a6-e88c2dad23c0.png)

## Motivation

It's a lot harder to read uncolored code

## Describe alternatives you've considered

Inline-git-diff is the closest alternative, however it doesn't work well for viewing diffs of old commits or diffs between two branches. I've looked for forks, and other community extensions.

## Additional context

I have some experience with Textmate/Tree sitter parsers,  tokenization, and colors. I'd like contribute to a (probably already-existing) discussion around this feature (colorized diffing in general).

**Comment by LPTK** [comment]
For me, the most important feature is that the diff should highlight the parts of the lines which have really changed, as in your second screenshot.

I cannot live without it, as it allows me to very quickly see exactly what changed. The current highlighting of the entire block in a uniform color is not helpful at all. I've been stuck using gitx precisely because most other git tools fail to do this right.

Some tools (like Github Desktop) do it for single lines of change, but not really for blocks of change. VSCode does it perfectly for blocks, but currently can't fold unchanged lines, making reviewing and staging diffs very clumsy.

**Comment by the-j0k3r** [comment]
@jeff-hykin 

What syntax-theme are you using, is this reproducible say when you use the syntax themes shipped with Atom?

Also, Im unsure this issue belongs here or rather in the syntax-themes repo https://github.com/atom?q=syntax&type=&language=

Trying around with different built in syntax themes this doesnt exist so it would have to be added in all supported syntax themes ideally.

**Comment by jeff-hykin** [comment]
The same issue occurs with the default themes.

And, to be clear the issue is that the Textmate scopes (or equivalent tree sitter scopes) are not given to the diffed code. The show cursor scopes just returns "text" instead of having a meaningfully parsed value

**Comment by smashwilson** [comment]
Closing as a duplicate of #1357.
